server/schedule/scheduler.go

```diff
@@ -107,12 +108,7 @@ func RegisterSliceDecoderBuilder(typ string, builder ConfigSliceDecoderBuilder)
 		log.Fatal("duplicated scheduler", zap.String("type", typ))
 	}
 	schedulerArgsToDecoder[typ] = builder
-}
-
-// IsSchedulerRegistered check where the named scheduler type is registered.
-func IsSchedulerRegistered(name string) bool {
-	_, ok := schedulerMap[name]
-	return ok
+	config.RegisterScheduler(typ)
```

Comment:
To avoid `config` depend on `scheduler`. I have to make 2 maps, one is in `scheduler` for creating schedulers, one is in `config` to validate configurations. So the register process should update 2 places at the same time.